frontend/packages/component-library/src/video/Video.tsx

```diff
@@ -0,0 +1,137 @@
+import {useState, useEffect, HTMLAttributes} from 'react';
+import classNames from 'classnames';
+
+import moduleStyles from './video.module.scss';
+import LinkButton from '../button/LinkButton';
+import FontAwesomeV6Icon from '@/fontAwesomeV6Icon';
+import {BodyTwoText, BodyThreeText, Figcaption, StrongText} from '@/typography';
+
+import {checkIfURLIsBlocked} from '@/common/helpers';
+
+export interface VideoProps extends HTMLAttributes<HTMLElement> {
+  /** Video title */
+  videoTitle?: string;
+  /** Video YouTube ID */
+  youTubeId?: string;
+  /** Video fallback */
+  videoFallback?: string;
+  /** Show caption */
+  showCaption?: boolean;
+  /** Label for Download button */
+  downloadLabel?: string;
+  /** Error heading for placeholder  */
+  errorHeading?: string;
+  /** Error body for placeholder */
+  errorBody?: string;
+  /** Video custom className */
+  className?: string;
+}
+
+/**
+ * ### Production-ready Checklist:
+ * * (✘) implementation of component approved by design team;
+ * * (✘) has storybook, covered with stories and documentation;
+ * * (✘) has tests: test every prop, every state and every interaction that's js related;
+ * * (see ./__tests__/Video.test.tsx)
+ * * (✘) passes accessibility checks;
+ *
+ * ###  Status: ```WIP```
+ *
+ * Design System: Video Component.
+ * This component is used to display a video from YouTube with a fallback HTML video player,
+ * and the option to download it, if an externally hosted fallback is provided.
+ * The video can also be displayed with a caption and works with responsive screen sizes.
+ */
+
+export const Video: React.FC<VideoProps> = ({
+  youTubeId,
+  videoTitle,
+  videoFallback,
+  showCaption,
+  downloadLabel,
+  errorHeading,
+  errorBody,
+  className,
+  ...HTMLAttributes
+}: VideoProps) => {
+  const [isYouTubeBlocked, setIsYouTubeBlocked] = useState(false);
+  const posterThumbnail = `//i.ytimg.com/vi/${youTubeId}/mqdefault.jpg`;
+
+  // Check to see if YouTube is blocked.
+  // If it is, we'll use the fallback video player.
+  useEffect(() => {
+    const checkYouTubeUrls = async () => {
+      const isYouTubeBlocked = await checkIfURLIsBlocked(
+        'https://www.youtube.com/favicon.ico',
+      );
+      const isYouTubeNoCookieBlocked = await checkIfURLIsBlocked(
+        'https://www.youtube-nocookie.com/favicon.ico',
+      );
+
+      setIsYouTubeBlocked(isYouTubeBlocked || isYouTubeNoCookieBlocked);
+    };
+
+    checkYouTubeUrls();
+  }, []);
```

Comment:
supepr nit: I wonder if we should refactor it slightly to something like:

```suggestion
useEffect(() => {
  const checkIfYoutubeIsBlocked = async () => {
      const [isYouTubeBlocked, isYouTubeNoCookieBlocked] = await Promise.all([
        checkIfURLIsBlocked('https://www.youtube.com/favicon.ico'),
        checkIfURLIsBlocked('https://www.youtube-nocookie.com/favicon.ico'),
      ]);

      setIsYouTubeBlocked(isYouTubeBlocked || isYouTubeNoCookieBlocked);
  };

  checkIfYoutubeIsBlocked();
}, []);
```

And maybe even move `checkIfYoutubeIsBlocked` definition to helpers as well?

Reply:
Does it make sense to move it to helpers since it's super specific to this component?

Reply:
Yeah, so initially I was thinking to propose to just move this function in same file, but before component definition so that it will only be created once instead of being created every time we render a video component.
Then I thought that, oh, this function checks if youtube is blocked, that's cool functionality! It might be used only in one place right now, but it might be theoretically helpful to someone for outside of the `Video` component context

Reply:
Just added this [here](https://github.com/code-dot-org/code-dot-org/pull/63435/commits/44aafdc7c9cf499a9c38997bfa5691b1bbf0005c), can you take a look before I merge?

Reply:
LGTM! Thank you!